### ISSUE TITLE
Objective value is wrong

### ISSUE THREAD

**Issue by projekter** [oxfordcontrol/Clarabel.jl#146]
I have a small quadratic program with linear and semidefinite constraints. The optimal objective value is -130.

```julia
using LinearAlgebra, SparseArrays, Clarabel
P = sparse([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 3, 4, 5, 6, 7, 8, 9, 10, 11, 4, 5, 6, 7, 8, 9, 10, 11, 5, 6, 7, 8, 9, 10, 11, 6, 7, 8, 9, 10, 11, 7, 8, 9, 10, 11, 8, 9, 10, 11, 9, 10, 11, 10, 11, 11], [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 5, 5, 5, 6, 6, 6, 6, 6, 6, 7, 7, 7, 7, 7, 8, 8, 8, 8, 9, 9, 9, 10, 10, 11], [2.0, 4.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 2.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.4444444444444444, 0.0, 0.0, 0.0, 0.0, -0.8888888888888887, 0.0, 0.0, 0.0, 0.44444444444444464, 0.0, 0.0, 0.0, 0.0, 0.0, -0.4444444444444443, 0.0, 1.0564784053156147, -1.3343410355945482, 0.0, 0.0, 0.28190303568898906, 0.0, -0.6013931428031767, 1.5282392026578073, 0.0, 0.0, 0.19933554817275745, 0.0, -0.42524916943521596, 1.1111111111111112, 0.0, 0.0, 0.0, 0.0, 0.4444444444444444, 0.0, 0.0, 0.0, 1.5282392026578073, 0.0, 0.14617940199335547, 1.1111111111111112, 0.0, 0.7774086378737541], 11, 11);
q = [-1.0, -1.0, -16.81342790821346, -0.0, 27.86611507785657, 20.704318936877076, 0.6285393610547088, 15.399214345840367, 6.132890365448506, -0.0, 15.049833887043189];
A = sparse([1, 12, 2, 12, 3, 5, 8, 4, 12, 6, 9, 7, 12, 10, 11, 12], [1, 1, 2, 2, 3, 4, 5, 6, 6, 7, 8, 9, 9, 10, 11, 11], [-1.0, 1.0, -1.0, 1.0, -1.0, -1.0, -1.0, -1.0, 1.0, -1.0, -1.0, -1.0, 1.0, -1.0, -1.0, 1.0], 12, 11);
b = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 10.0];
cones = [Clarabel.NonnegativeConeT(1), Clarabel.PSDTriangleConeT(4), Clarabel.NonnegativeConeT(1)];
settings = Clarabel.Settings();
solver = Clarabel.Solver();
Clarabel.setup!(solver, P, q, A, b, cones, settings);
result = Clarabel.solve!(solver)
```
Now, the result status is `SOLVED`, however, Clarabel reports a better optimal objective value than possible:
```julia
julia> result.obj_val
-146.0668053087124
julia> dot(result.x, P, result.x)/2 + dot(result.x, q)
-129.2779437022457
```
So the actual solution that is returned has a different objective value that the objective value in the result object. Additionally, the duality gap is reported to be `9.79e-9`. For this small a gap, even the "actual" result seems to be still a bit far from the optimal value. Mosek reports `-130.00045750377342`, Mathematica `-130.00045729259804`...

**Comment by goulart-paul** [comment]
Yes, that is not good.   Thank you for submitting the issue.

@yuwenchen95 : maybe this is an issue with calculating objectives with scaled PSD cones?   That's a guess though.

**Comment by projekter** [comment]
Sorry for the late reply, I didn't get a notification of your edit and only saw it when I manually checked the issue.
I don't use JuMP/MOI; the problems are formulated directly in the respective solver interfaces. Unfortunately, I don't exactly know which of the many test problems I used to produce this particular example, so I tried to reconstruct it (luckily I found and old stored Mosek task that gives 130 as objective) and found out that the blame goes to me completely.

The `P` I gave above is lower-triangular, whereas I now read in the manual that it should be the upper triangle of the relevant matrix. This explains the discrepancy of objective and calculated objective.
Plus, even if I transpose `P`, there is some confusion in my example about the off-diagonal scaling. For the given `A`, which makes my `x[2:end]` to be the scaled lower triangles of the PSD matrix, the correct `P` should actually be
```Julia
sparse([1, 1, 2, 3, 4, 5, 5, 6, 7, 3, 8, 5, 6, 9, 4, 10, 5, 6, 9, 11], [1, 2, 2, 3, 4, 5, 6, 6, 7, 8, 8, 9, 9, 9, 10, 10, 11, 11, 11, 11], [2.0, 2.0, 2.0, 0.444444, 0.444444, 1.05648, -0.667171, 1.52824, 1.11111, -0.444444, 0.444444, 0.140952, 0.0996678, 1.52824, -0.222222, 1.11111, -0.300697, -0.212625, 0.0730897, 0.777409], 11, 11)
```
And now result is indeed correct - so I'm very sorry about this noise, I don't know what went wrong the last time.
